MiniApp/Classes/core/JavascriptBridge/MiniAppScriptMessageHandler.swift

```diff
@@ -55,7 +55,13 @@ internal class MiniAppScriptMessageHandler: NSObject, WKScriptMessageHandler {
             requestDevicePermission(requestParam: requestParam, callbackId: callbackId)
         case .getCurrentPosition:
             locationManager = LocationManager(enableHighAccuracy: requestParam?.locationOptions?.enableHighAccuracy ?? false)
-            getCurrentPosition(callbackId: callbackId)
+            locationManager?.updateLocation {[weak self] result in
+                switch result {
+                case .success:
+                    self?.getCurrentPosition(callbackId: callbackId)
+                default: self?.executeJavaScriptCallback(responseStatus: .onError, messageId: callbackId, response: getMiniAppErrorMessage(MASDKCustomPermissionError.userDenied))
+                }
+            }
```

Comment:
Can we move this to a separate function?

Reply:
of course